### DISCUSSION TITLE
PCAP capture

### DISCUSSION THREAD

**Discussion by arman-maghsoudnia** [srsran/srsRAN_Project#475] [General Help] [answered]
Hi,
I was using the packets captured inside the RLC layer using the configuration that you provide

> rlc_filename: /tmp/gnb_rlc.pcap       # Optional TEXT (tmp/gnb_rlc.pcap). Path for RLC PCAPs. 
  rlc_rb_type: all                      # Optional TEXT. Sets the RLC PCAP RB type. Supported: [all, srb, drb].
  rlc_enable: enable                # Optional BOOL (false). Enable/disable RLC packet capture.  

I was wondering at which point these packets are captured in the RLC layer. I mean are they captured at the beginning of RLC layer when the packets enter the RLC layer or at the end of the RLC layer when the packets are ready to be sent into the MAC layer. Knowing this is very important for me because I want to know whether the process is already done or not.  Could you please clarify this? 
I will attach two screenshots of the packets I captured in the RLC layer. 

![image](https://github.com/srsran/srsRAN_Project/assets/93431904/1d9fc0d2-11a8-4e0c-a0d6-e822e76e69b6)

![image](https://github.com/srsran/srsRAN_Project/assets/93431904/b4b75df5-fac9-408f-8c5e-290b84d3851f)

**Comment by robertfalkenberg** [comment, marked as answer]
The RLC PCAP captures "RLC PDUs" which are exchanged between MAC and RLC.

To properly display and dissect RLC PCAPs with wireshark, you need to build latest wireshark from source as the necessary changes will most likely be included in future wireshark v4.3.x

**Reply by robertfalkenberg** [reply]
Here is an example how it should look like:

![image](https://github.com/srsran/srsRAN_Project/assets/86455551/50d31efa-9229-4322-9353-1654b1226764)

**Reply by arman-maghsoudnia** [reply]
Thanks. Could you please update this page: https://docs.srsran.com/projects/project/en/latest/user_manuals/source/outputs.html
There is no information about RLC there right now. And can you here tell me the wireshark setting to get the results as the screenshot you sent?
And do you have a plan to support PCAP capture at other layers like PHY and PDCP and SDAP?

**Reply by robertfalkenberg** [reply]
Please follow the instructions for MAC
https://docs.srsran.com/projects/project/en/latest/user_manuals/source/outputs.html#mac

and also activate `rlc_nr_udp`.

We will update the manual once wireshark is released and no build from source is required.

In the mean time, you can also use MAC PCAP, which will give you similar outputs, as wireshark will automatically dissect the MAC PDUs into RLC PDUs and forth.

There are currently no plans for PCAPs at PDCP and SDAP layer as they can be already covered by RLC and MAC PCAPs.

**Comment by arman-maghsoudnia** [comment]
I edited the discussion name so I also ask my questions about MAC PCAP capture here as well. I did the MAC layer capture and configured the wireshark setting as mentioned in this guide: https://docs.srsran.com/projects/project/en/latest/user_manuals/source/outputs.html
But I only can see the Scheduling decision packets in the wireshark (I attach a screenshot). I also want to see the actual data being forwarded from the MAC layer to the PHY layer. Do you know why I cannot see this data and I only see the scheduling decisions data? 

The first screenshot is from the beginning of the capture:
![image](https://github.com/srsran/srsRAN_Project/assets/93431904/ed355cf5-a0ac-4601-b8d3-9202a89f76f2)

The second screenshot is the middle of the capture. As you can see the info section indicates that only UL and DL scheduling packets are captured and I cannot see the other data going to the PHY layer. 
![image](https://github.com/srsran/srsRAN_Project/assets/93431904/f773328d-4468-405f-a809-36a437f6e228)

**Reply by robertfalkenberg** [reply]
Also enable this:

![image](https://github.com/srsran/srsRAN_Project/assets/86455551/57225135-ce5f-4bbd-b768-e77132beeea9)
